app/scripts/controllers/bridge/bridge-controller.ts

```diff
@@ -319,4 +340,25 @@ export default class BridgeController extends StaticIntervalPollingController<
       'NetworkController:getSelectedNetworkClient',
     );
   }
+
+  /**
+   *
+   * @param contractAddress - The address of the ERC20 token contract
+   * @param walletAddress - The address of the wallet
+   * @param chainId - The hex chain ID of the bridge network
+   * @returns The atomic allowance of the ERC20 token contract
+   */
+  getBridgeERC20Allowance = async (
+    contractAddress: string,
+    walletAddress: string,
+    chainId: Hex,
+  ): Promise<string> => {
+    const web3Provider = new Web3Provider(this.#provider);
```

Comment:
We can also get the current provider with  `const provider = this.#getSelectedNetworkClient()?.provider;` . This way the provider doesn't need to be passed in to the constructor

Reply:
Done in [af0c5a3](https://github.com/MetaMask/metamask-extension/pull/27262/commits/af0c5a30c37e8cba64c32489b889b7efd1cf8110)